flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/Execution.java

```diff
@@ -1133,7 +1128,10 @@ private void processFail(
         checkState(transitionState(current, FAILED, t));
 
         // success (in a manner of speaking)
-        this.failureCause = t;
+        if (t != null) {
+            // we only set the failureCause if an error is passed (see FLINK-21376)
+            this.failureCause = Optional.of(new ErrorInfo(t, getStateTimestamp(FAILED)));
+        }
```

Comment:
Can't we say that if `t == null`, then we create an unknown failure cause? `new FlinkException("Unknown cause for Execution failure. This might be caused by FLINK-21376")`?